src/runtime/policy/ingress-validation.ts

```diff
@@ -298,21 +298,9 @@ export class IngressValidation {
         // should make sure that the `canWriteSuperset` is consistent with the
         // `canReadSubset` of the type variable, which can be achieved by using
         // the `restrictTypeRanges` method.
-        //
-        // TODO(b/175169555): Schema.intersect has a bug with respect to inline
-        // entites. The following code works around the bug and when it is fixed,
-        // it should be simplified as follows:
-        //   const canWriteSuperset = maxReadType.restrictTypeRanges(canReadSubset);
-        let newCanWriteSuperset = null;
-        if (maxReadType.isAtLeastAsSpecificAs(canReadSubset)) {
-          newCanWriteSuperset = canReadSubset;
-        } else if (canReadSubset.isAtLeastAsSpecificAs(maxReadType)) {
-          newCanWriteSuperset = maxReadType;
-        } else {
-          newCanWriteSuperset = maxReadType.restrictTypeRanges(canReadSubset);
-        }
+        const canWriteSuperset = maxReadType.restrictTypeRanges(canReadSubset);
         return TypeVariable.make(
-          '', newCanWriteSuperset, canReadSubset, typeVar.resolveToMaxType);
+          '', canWriteSuperset, canReadSubset, typeVar.resolveToMaxType);
```

Comment:
Sorry, I think the comment was off.  Can you use  `newCanWriteSuperset` here and above to indicate it is a recomputed value?

Reply:
Ah, sure thing. Thanks for the clarity tip.